src/requests.rs

```diff
@@ -194,9 +272,84 @@ pub async fn fetch_l1_data(
         new_root,
         bootloader_hash: *bootloader_code_hash.as_fixed_bytes(),
         default_aa_hash: *default_aa_code_hash.as_fixed_bytes(),
+        prev_batch_commitment,
+        curr_batch_commitment,
     };
 
-    Ok(result)
+    Ok((result, aux_output))
+}
+
+pub async fn fetch_proof_from_l1(
```

Comment:
add comments - mention what it returns.

Also - might be easier to return the "Result" - rathern than unwraps (but that's the nit)